src/cpu/x64/gemm/gemm_info.cpp

```diff
@@ -878,12 +876,8 @@ void gemm_info_t<a_t, b_t, c_t>::jit_init(void) {
     constexpr bool is_bf16 = data_traits<a_t>::data_type == data_type::bf16;
     bool doAlpha1 = this->alpha != 1.0f && is_bf16 ? no_alpha1 : do_alpha1;
 
-    {
-        // NOTE: This lock may not be needed at all as writes to copy_a_kern
-        // (and others) are protected within std::call_once(). The lock is added
-        // only to fix warnings reported by clang TSAN about a data race in
-        // this code block.
-        std::lock_guard<std::mutex> g(kern_mutex);
+    std::once_flag post_init_flag;
```

Comment:
This will probably need to be `static` to actually guarantee it's only called once.

Reply:
nice catch. addressed